catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/ImportSnapshotWorker.java

```diff
@@ -107,22 +93,22 @@ private EntitySnapshotObj.Builder importIcebergTable(
     // snapshot!=null means that we already have the snapshot (via a committing operation - like a
     // table update) and do not need to import it but can just store it.
     if (snapshot == null) {
-      IcebergTableMetadata tableMetadata;
       StorageUri metadataLocation = StorageUri.of(content.getMetadataLocation());
+      NessieTable table;
+      IcebergTableMetadata tableMetadata;
       try {
         InputStream input = taskRequest.objectIO().readObject(metadataLocation);
         if (metadataLocation.requiredPath().endsWith(".gz")
             || metadataLocation.requiredPath().endsWith(".gz.metadata.json")) {
           input = new GZIPInputStream(input);
         }
         tableMetadata = IcebergJson.objectMapper().readValue(input, IcebergTableMetadata.class);
-      } catch (IOException e) {
-        throw new IOException(
+        table = entityObjForContent(content, tableMetadata, entityObjId);
```

Comment:
Any specific reason why you moved this inside the try-block? (same for view below)

Reply:
Just to wrap it with a `RuntimeException` mentioning the view/table name for `IcebergErrorMapper.message()` - makes user-level errors easier to interpret (and the new tests check for that).